frontend/public/components/cluster-settings/gitlab-idp-form.tsx

```diff
@@ -90,20 +97,37 @@ export class AddGitLabPage extends PromiseComponent<{}, AddGitLabPageState> {
     return this.handlePromise(addIDP(oauth, idp));
   }
 
+  dryCreateOAuth(oauth): Promise<K8sResourceKind> {
```

Comment:
I'd make this an optional flag on the existing `addIDP` helper so that the request with be the same as the real request